ncclient/transport/ssh.py

```diff
@@ -149,149 +163,84 @@ def _parse10(self):
                 self._parsing_pos10 = 0
 
     def _parse11(self):
-        logger.debug("parsing netconf v1.1")
-        expchunksize = self._expchunksize
-        curchunksize = self._curchunksize
-        idle, instart, inmsg, inbetween, inend = range(5)
-        state = self._parsing_state11
-        inendpos = self._inendpos
-        num_list = self._size_num_list
-        MAX_STARTCHUNK_SIZE = 12 # \#+4294967295+\n
-        pre = 'invalid base:1:1 frame'
-        buf = self._buffer
-        buf.seek(self._parsing_pos11)
-        message_list = self._message_list # a message is a list of chunks
-        chunk_list = []   # a chunk is a list of characters
-
-        should_recurse = False
-
+        
+        """Messages are split into chunks. Chunks and messages are delimited
+        by the regex #RE_NC11_DELIM defined earlier in this file. Each time we
+        get called here either a chunk delimiter or an end-of-message delimiter
+        should be found. If it's not, a framing error will be raised."""
```

Comment:
Comment is misleading. If I'm reading this right, we get a NetconfFramingError only if we *do* find the delimiter but it's not where we expect it (at the start of the buffer). If we don't find the delimiter at all, then this function is a no-op. Is that correct?

Reply:
Yes. Comment improved:

```
        """Messages are split into chunks. Chunks and messages are delimited
        by the regex #RE_NC11_DELIM defined earlier in this file. Each
        time we get called here either a chunk delimiter or an
        end-of-message delimiter should be found iff there is enough
        data. If there is not enough data, we will wait for more. If
        data is found in the wrong place, a #NetconfFramingError will
        be raised."""
```

Reply:
Comment improved:

```
        """Messages are split into chunks. Chunks and messages are delimited
        by the regex #RE_NC11_DELIM defined earlier in this file. Each
        time we get called here either a chunk delimiter or an
        end-of-message delimiter should be found iff there is enough
        data. If there is not enough data, we will wait for more. If a
        delimiter is found in the wrong place, a #NetconfFramingError
        will be raised."""
```